File: 08-repeticao-arranjos-e-conjuntos/exemplos/indice-maximo.cpp
#include "bscpp.hpp"
#include <vector>

using namespace std;

// Encontra o índice da primeira ocorrência do valor máximo de valores.
int indice_maximo(vector<int> valores)
{
    int imax = 0;
    for (int i = 1; i < valores.size(); i = i + 1) {
        if (valores[i] > valores[imax]) {
            imax = i;
        }
    }
    return imax;
}

examples
{
    check_expect(indice_maximo({5}), 0);
    check_expect(indice_maximo({5, 5}), 0);
    check_expect(indice_maximo({5, 7, 5}), 1);
    check_expect(indice_maximo({5, 7, 5, 8}), 3);
}

int main()
{
    run_tests();
}

File: 07-outras-formas-de-repeticao/exemplos/indice_maximo.py
def indice_maximo(lst: list[int]) -> int:
    '''
    Encontra o índice da primeira ocorrência do valor máximo de *lst*.
    Requer que *lst* seja não vazia.

    Exemplos
    >>> indice_maximo([5])
    0
    >>> indice_maximo([5, 6])
    1
    >>> indice_maximo([5, 6, 6])
    1
    >>> indice_maximo([5, 6, 6, 8])
    3
    '''
    assert len(lst) != 0
    imax = 0
    for i in range(1, len(lst)):
        if lst[i] > lst[imax]:
            imax = i
    return imax
